### PR TITLE
bluetooth: keys_br: Improve bt_foreach_bond

### PR DESCRIPTION
**Pull request by lylezhu2012**
The BR Keys cannot be scanned by function bt_foreach_bond.

Add function bt_foreach_bond_br for br.

The function bt_foreach_bond_br will be called by bt_foreach_bond if the BR is enabled.

### CODE REVIEW

File: subsys/bluetooth/host/classic/keys_br.c [hunk 1]
```diff
@@ -146,6 +146,26 @@ void bt_keys_link_key_store(struct bt_keys_link_key *link_key)
 	}
 }
 
+void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info,
+					   void *user_data), void *user_data)
```
Comment (Thalley) [inline subsys/bluetooth/host/classic/keys_br.c:150]: Indentation is off here

File: subsys/bluetooth/host/classic/keys_br.c [hunk 2]
```diff
@@ -146,6 +146,26 @@ void bt_keys_link_key_store(struct bt_keys_link_key *link_key)
...
+{
+	int i;
+
+	__ASSERT_NO_MSG(func != NULL);
+
+	for (i = 0; i < ARRAY_SIZE(key_pool); i++) {
+		struct bt_keys_link_key *key = &key_pool[i];
```
Comment (Thalley) [inline subsys/bluetooth/host/classic/keys_br.c:157]:
```suggestion
		const struct bt_keys_link_key *key = &key_pool[i];
```

File: subsys/bluetooth/host/keys.h [hunk 3]
```diff
@@ -215,7 +215,8 @@ struct bt_keys_link_key *bt_keys_find_link_key(const bt_addr_t *addr);
 void bt_keys_link_key_clear(struct bt_keys_link_key *link_key);
 void bt_keys_link_key_clear_addr(const bt_addr_t *addr);
 void bt_keys_link_key_store(struct bt_keys_link_key *link_key);
-
+void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info,
+					   void *user_data), void *user_data);
```
Comment (Thalley) [inline subsys/bluetooth/host/keys.h:219]: Ditto with indentation

**Review by Thalley — CHANGES_REQUESTED** [review]
a few minor things

File: subsys/bluetooth/host/classic/keys_br.c [hunk 4]
```diff
@@ -146,6 +146,26 @@ void bt_keys_link_key_store(struct bt_keys_link_key *link_key)
...
 }
 
+void bt_foreach_bond_br(void (*func)(const struct bt_bond_info *info,
+					   void *user_data), void *user_data)
+{
+	int i;
+
+	__ASSERT_NO_MSG(func != NULL);
+
+	for (i = 0; i < ARRAY_SIZE(key_pool); i++) {
```
[the remark this answers is not in the document]
Comment (lylezhu2012) [reply subsys/bluetooth/host/classic/keys_br.c]: Updated

File: subsys/bluetooth/host/classic/keys_br.c (hunk 2 above)
Comment (lylezhu2012) [reply subsys/bluetooth/host/classic/keys_br.c]: Updated

File: subsys/bluetooth/host/keys.h (hunk 3 above)
Comment (lylezhu2012) [reply subsys/bluetooth/host/keys.h]: Updated

File: subsys/bluetooth/host/classic/keys_br.c (hunk 1 above)
Comment (lylezhu2012) [reply subsys/bluetooth/host/classic/keys_br.c]: Updated

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**